paddle/fluid/framework/ir/graph_helper.cc

```diff
@@ -99,6 +100,63 @@ std::vector<ir::Node *> TopologySortOperations(const Graph &graph) {
   return ret;
 }
 
+std::vector<ir::Node *> SortOperationsInSequence(const Graph &graph) {
+  PADDLE_ENFORCE(graph.Has(kAllOpDescs),
+                 "Graph has no attribute of kAllOpDescs.");
+  auto &ops = graph.Get<std::vector<OpDesc *>>(kAllOpDescs);
```

Comment:
I have tried that way, but I found two killer issue hard to solve.
1. the current graph do not have the order information. Recover the topology sort from an unordered graph, it will have more than one candidate sequence. The memory optimize heavily depend on the order of the op sequence, so each sequence will have a totally different reuse result. It's hard to determine which candidate sequence is better, unless we will exhaustive all the possible sequences and do a greedy search. 
And the same time, I found the op sequence same with the python logic is "almost" the best one. Because our python version memory optimize just use it, it's hard to overcome the python version. 
2. Currently, there are only two kinds of pass will create new op descs. 
     2.1 fused kernel pass. It will substitute the old subgraph with a new one. we can mark that subgraph in sequence.
     2.2 merge_multi_batch_pass, as you can see, I also modify the merge_multi_batch_pass opdescs related code. 
To be honest, use the cached opdesc sequence is not a final solution. I'm still thinking how to make it reasonable.
Exhaustive all the possible sequences and do a greedy search maybe one of my option, but it will not be included in 1.2 version.